laws/shared/src/main/scala/cats/laws/discipline/Arbitrary.scala

```diff
@@ -52,4 +52,27 @@ object arbitrary {
 
   implicit def appFuncArbitrary[F[_], A, B](implicit F: ArbitraryK[F], B: Arbitrary[B], FF: Applicative[F]): Arbitrary[AppFunc[F, A, B]] =
     Arbitrary(F.synthesize[B].arbitrary.map(fb => Func.appFunc[F, A, B](_ => fb)))
+
+  implicit def streamingArbitrary[A](implicit A: Arbitrary[A]): Arbitrary[Streaming[A]] =
+    Arbitrary(Gen.listOf(A.arbitrary).map(Streaming.fromList(_)))
+
+  // TODO: it would be better to do this recursively, i.e. more like:
+  //
+  // Gen.oneOf(
+  //   for { a <- arbitrary[A]; s <- arbitrary[F[StreamingT[F, A]]] } yield cons(a, s),
+  //   for { s <- arbitrary[F[StreamingT[F, A]]] } yield wait(s),
+  //   const(StreamingT.empty[F, A]))
+  //
+  // Howver, getting this right with Scalacheck (and avoiding SOEs) is
+  // somewhat fiddly, so this will have to do for now.
+  //
+  // The max possible size of a StreamingT instance (n) will result in
+  // instances of up to n^3 in length when testing flatMap
+  // composition. The current value (8) could result in streams of up
+  // to 512 elements in length. Thus, since F may not be stack-safe,
+  // we want to keep n relatively small.
```

Comment:
workaround in scalaz https://github.com/scalaz/scalaz/commit/450ed685ac52ca6fe63